### PR TITLE
elasticsearch: increase settings

### PR DESCRIPTION
**Pull request by hectorj2f**
We are checking if these new settings would tackle recent issues on soak

### CODE REVIEW

File: templates/elasticsearch.yaml [hunk 1]
```diff
@@ -25,18 +25,18 @@ spec:
...
           # need more cpu upon initialization, therefore burstable class
           limits:
-            cpu: 1000m
-            memory: 4096Mi
+            cpu: 4000m
+            memory: 8192Mi
           requests:
             cpu: 250m
-            memory: 2048Mi
+            memory: 8000Mi
```
Comment (jieyu) [inline templates/elasticsearch.yaml]: I feel we should simply disallow bursting for memory. Memory bursting relies on OOM score adjustment, which is not very reliable. I would rather prefer a more deterministic behavior on this.

i.e., change this to 8192Mi

File: templates/elasticsearch.yaml [hunk 2]
```diff
@@ -25,18 +25,18 @@ spec:
...
+        heapSize: 3072m
         resources:
           # need more cpu upon initialization, therefore burstable class
           limits:
-            cpu: 1000m
-            memory: 4096Mi
+            cpu: 4000m
+            memory: 8192Mi
           requests:
             cpu: 250m
```
Comment (jieyu) [inline templates/elasticsearch.yaml]: Consider change requests to 1000m

File: templates/elasticsearch.yaml [hunk 3]
```diff
@@ -25,18 +25,18 @@ spec:
         resources:
           # need more cpu upon initialization, therefore burstable class
           limits:
-            cpu: 1000m
+            cpu: 2000m
             memory: 2048Mi
           requests:
             cpu: 250m
```
Comment (jieyu) [inline templates/elasticsearch.yaml]: consider change request to 500m

File: templates/elasticsearch.yaml [hunk 4]
```diff
@@ -25,18 +25,18 @@ spec:
...
             memory: 1536Mi
```
Comment (jieyu) [inline templates/elasticsearch.yaml]: See below. I'd prefer not bursting memory. i.e., set it to 2048Mi

File: templates/elasticsearch.yaml (hunk 1 above)
Comment (hectorj2f) [reply templates/elasticsearch.yaml]: Disallowing bursting goes against the settings in our ES installations for Kubernetes which recommend
```
In terms of Memory resources you should make sure that you follow that equation:

${role}HeapSize < ${role}MemoryRequests < ${role}MemoryLimits
```

Comment (hectorj2f) [reply templates/elasticsearch.yaml]: For instance in: https://github.com/helm/charts/tree/master/stable/elasticsearch

File: templates/elasticsearch.yaml (hunk 4 above)
Comment (hectorj2f) [reply templates/elasticsearch.yaml]: same as above: https://github.com/mesosphere/kubeaddons-configs/pull/16#discussion_r285500317

**Comment by hectorj2f** [comment]
@jieyu Could you review again this PR and what you think about my last comments ?

File: templates/elasticsearch.yaml (hunk 1 above)
Comment (shaneutt) [reply templates/elasticsearch.yaml]: `8000` seems very high when our heapsize is `3072`, could we not adjust our request down to `4096` to significantly improve the pods ability to be scheduled?

Also it does seem like we would want to keep bursting available, as that will provide memory wiggle-room while still giving us a better baseline to get scheduled.

Comment (jieyu) [reply templates/elasticsearch.yaml]: OK, that's fair. I remember that we saw memory usage spike to 7G+ in soak. So the limit has to be 8G. The request can be lower.

I think it's OK to reduce memory request to 4608 (1.5x the heap size) given the heap size is 3072.

Thoughts?

File: templates/elasticsearch.yaml (hunk 4 above)
Comment (jieyu) [reply templates/elasticsearch.yaml]: OK. that's fair! Thanks for the pointer.

File: templates/elasticsearch.yaml (hunk 1 above)
Comment (shaneutt) [reply templates/elasticsearch.yaml]: Sounds good :+1:

Comment (hectorj2f) [reply templates/elasticsearch.yaml]: Done

Comment (hectorj2f) [reply templates/elasticsearch.yaml]: IMHO, we won't see an impact until we start using it, and see if that value is enough or not

**Comment by hectorj2f** [comment]
Done

**Comment by hectorj2f** [comment]
@jieyu @shaneutt I've addressed your comments, please re-review :)

**Pull request merged**